src/TorchVision/AutoAugment.cs

```diff
@@ -101,38 +107,206 @@ protected enum opType {
+
+                var (transform_id, probs, signs) = this.get_params(this.policies.Count);
+
+                var op_meta = this.augmentation_space(10, (height, width));
+
+                var (policy1, policy2) = policies[transform_id];
+                foreach (var (i, (op_name, p, magnitude_id)) in new[]{(0, policy1), (1, policy2)})
+                {
+                    if (probs[i].ToDouble() <= p) {
+                        var (magnitudes, signed) = op_meta[op_name];
+                        var magnitude = magnitude_id != null ? magnitudes[magnitude_id].ToDouble() : 0.0;
+                        if (signed && signs[i].ToBoolean())
+                            magnitude *= -1.0;
+                        img = apply_op(img, op_name, magnitude, interpolation: this.interpolation, fill: this.fill);
+                    }
+                }
+
+                return img;
+            }
+
+            private (int, Tensor, Tensor) get_params(int transform_num)
+            {
+                var policy_id = torch.randint(0, transform_num, size: 1, generator: this.generator).ToInt32();
+                var probs = torch.rand(2, generator: this.generator);
+                var signs = torch.randint(0, 2, size: 2, generator: this.generator);
+                return (policy_id, probs, signs);
+            }
+
+            private Dictionary<opType, (Tensor, bool)> augmentation_space(int num_bins, (long height, long width) image_size)
+            {
+                return new Dictionary<opType, (Tensor, bool)> {
+                        { opType.ShearX, (torch.linspace(0.0, 0.3, num_bins), true)},
+                        { opType.ShearY, (torch.linspace(0.0, 0.3, num_bins), true)},
+                        { opType.TranslateX, (torch.linspace(0.0, 150.0 / 331.0 * image_size.width, num_bins), true)},
+                        { opType.TranslateY, (torch.linspace(0.0, 150.0 / 331.0 * image_size.height, num_bins), true)},
+                        { opType.Rotate, (torch.linspace(0.0, 30.0, num_bins), true)},
+                        { opType.Brightness, (torch.linspace(0.0, 0.9, num_bins), true)},
+                        { opType.Color, (torch.linspace(0.0, 0.9, num_bins), true)},
+                        { opType.Contrast, (torch.linspace(0.0, 0.9, num_bins), true)},
+                        { opType.Sharpness, (torch.linspace(0.0, 0.9, num_bins), true)},
+                        { opType.Posterize, (8 - (torch.arange(num_bins) / ((num_bins - 1) / 4)).round().@int(), false)},
+                        { opType.Solarize, (torch.linspace(255.0, 0.0, num_bins), false)},
+                        { opType.AutoContrast, (torch.tensor(0.0), false)},
+                        { opType.Equalize, (torch.tensor(0.0), false)},
+                        { opType.Invert, (torch.tensor(0.0), false)}
+                    };
+            }
+
+            private IList<((opType, double, int?), (opType, double, int?))> get_policies(AutoAugmentPolicy policy)
```

Comment:
For methods that are not a 1:1 correspondence to something in the PyTorch APIs, I try (but I sometimes forget myself) to stay with .NET naming conventions. The only reason for not using .NET conventions is to make it super-easy to move code from Python to C#, so when that's not a motivating reason, better to stick with .NET-style names.

Reply:
Yes.

Reply:
I'll leave this if thats ok. They are all private methods, would the `augmentation_space` functions need to be changed as well?